utils/releaser/src/ReleaseWorker/ReleaseCandidate/ValidateRequireFormatInComposerJsonReleaseWorker.php

```diff
@@ -0,0 +1,148 @@
+    /**
+     * @var \Symplify\MonorepoBuilder\Package\PackageNamesProvider
+     */
+    private $packageNamesProvider;
+
+    /**
+     * @param \Symplify\MonorepoBuilder\FileSystem\ComposerJsonProvider $composerJsonProvider
+     * @param \Symplify\MonorepoBuilder\FileSystem\JsonFileManager $jsonFileManager
+     * @param \Symplify\MonorepoBuilder\Package\PackageNamesProvider $packageNamesProvider
+     */
+    public function __construct(
+        ComposerJsonProvider $composerJsonProvider,
+        JsonFileManager $jsonFileManager,
+        PackageNamesProvider $packageNamesProvider
+    ) {
+        $this->composerJsonProvider = $composerJsonProvider;
+        $this->jsonFileManager = $jsonFileManager;
+        $this->packageNamesProvider = $packageNamesProvider;
+    }
+
+    /**
+     * @param \PharIo\Version\Version $version
+     * @return string
+     */
+    public function getDescription(Version $version): string
+    {
+        return 'Validate "require" and "require-dev" version format for all packages';
+    }
+
+    /**
+     * Higher first
+     * @return int
+     */
+    public function getPriority(): int
+    {
+        return 920;
+    }
+
+    /**
+     * @param \PharIo\Version\Version $version
+     */
+    public function work(Version $version): void
+    {
+        foreach ($this->composerJsonProvider->getRootAndPackageFileInfos() as $smartFileInfo) {
+            $jsonContent = $this->jsonFileManager->loadFromFileInfo($smartFileInfo);
+
+            $this->validateVersions($jsonContent, 'require', $smartFileInfo);
+            $this->validateVersions($jsonContent, 'require-dev', $smartFileInfo);
+        }
+
+        if ($this->isSuccessful) {
+            $this->symfonyStyle->success(Message::SUCCESS);
+        } else {
+            $this->confirm('Confirm all the requires are in the valid format');
+        }
+    }
+
+    /**
+     * @param mixed[] $jsonContent
+     * @param string $section
+     * @param \Symfony\Component\Finder\SplFileInfo $splFileInfo
+     */
+    private function validateVersions(array $jsonContent, string $section, SplFileInfo $splFileInfo): void
+    {
+        if (!isset($jsonContent[$section])) {
+            return;
+        }
+
+        foreach ($jsonContent[$section] as $packageName => $version) {
+            if ($this->shouldSkipPackageNameAndVersion($packageName, $version)) {
+                continue;
+            }
+
+            $this->symfonyStyle->warning(sprintf(
+                '"%s" file has invalid version format for "%s": "%s"',
+                $splFileInfo->getPathname(),
+                $packageName,
+                $version
+            ));
+
+            $this->isSuccessful = false;
+        }
+    }
+
+    /**
+     * @param string $packageName
+     * @param string $version
+     * @return bool
+     */
+    private function shouldSkipPackageNameAndVersion(string $packageName, string $version): bool
+    {
+        if (Strings::startsWith($packageName, 'ext-')) {
```

Comment:
does it make sense to merge these 3 conditions into one return `||` command ?
result would look like this
```
return Strings::startsWith($packageName, 'ext-') || Strings::startsWith($version, '^') || in_array($packageName, $this->packageNamesProvider->provide(), true);
```
is is possible in `nette` to do it like this?
```
return Strings::startsWith([$packageName, $version], ['ext-', '^']) || in_array($packageName, $this->packageNamesProvider->provide(), true);
```

Reply:
Why do you preffer it?